In [11]:
import pandas as pd
import numpy as np
import statistics

In [12]:
age_list = [14, 12, 11, 10, 8, 6, 8]

In [13]:
list_mean = np.mean(age_list)
list_median = np.median(age_list)
list_mode = statistics.mode(age_list)
list_variance = np.var(age_list)
list_std = np.std(age_list)
math_error = np.std(age_list) / np.sqrt(len(age_list)-1)

In [14]:
print("The mean of the data set is {}.".format(list_mean))
print("The median of the data set is {}.".format(list_median))
print("The mode of the data set is {}.".format(list_mode))
print("The variance of the data set is {}.".format(list_variance))
print("The standard deviation of the data set is {}.".format(list_std))
print("The standard error of our mean calculation is {}.".format(math_error))

The mean of the data set is 9.857142857142858.
The median of the data set is 10.0.
The mode of the data set is 8.
The variance of the data set is 6.408163265306122.
The standard deviation of the data set is 2.531435020952764.
The standard error of our mean calculation is 1.0334540197243194.


This data set above represents the ages of six children.  For this example I would use the mean as the measure of central tendency, because here we are working with a small data set and so the mean (average) is a good representation of the data set.  For the measure of variance, I would use the standard deviation.  The standard deviation of this data set is 2.73.  From looking at our data set, I see that the majority of the data points fall within or are close to the standard deviation.

In [15]:
age_list2 = [14, 12, 11, 10, 8, 7, 8]

In [16]:
list_mean2 = np.mean(age_list2)
list_median2 = np.median(age_list2)
list_mode2 = statistics.mode(age_list2)
list_variance2 = np.var(age_list2)
list_std2 = np.std(age_list2)
math_error2 = np.std(age_list2) / np.sqrt(len(age_list2)-1)

In [17]:
print("The mean of the data set is {}.".format(list_mean2))
print("The median of the data set is {}.".format(list_median2))
print("The mode of the data set is {}.".format(list_mode2))
print("The variance of the data set is {}.".format(list_variance2))
print("The standard deviation of the data set is {}.".format(list_std2))
print("The standard error of our mean calculation is {}.".format(math_error2))

The mean of the data set is 10.0.
The median of the data set is 10.0.
The mode of the data set is 8.
The variance of the data set is 5.428571428571429.
The standard deviation of the data set is 2.32992949004287.
The standard error of our mean calculation is 0.951189731211342.


If the youngest child has a birthday, and we increase his/her age from 6 to 7 yrs old, then the mean (average) of the data set increases alittle from 9.85 to 10.  The median of the data does not change.  The variance decreases alittle from 6.4 to 5.43 and the standard deviation decreases from 2.73 to 2.52, because we increased the age of the youngest child from 6 to 7, pushing the aggregate data got closer to the mean.

In [18]:
age_list3 = [14, 12, 11, 10, 8, 7, 1]

In [19]:
list_mean3 = np.mean(age_list3)
list_median3 = np.median(age_list3)
list_variance3 = np.var(age_list3)
list_std3 = np.std(age_list3)
math_error3 = np.std(age_list3) / np.sqrt(len(age_list3)-1)

In [20]:
print("The mean of the data set is {}.".format(list_mean3))
print("The median of the data set is {}.".format(list_median3))
print("The variance of the data set is {}.".format(list_variance3))
print("The standard deviation of the data set is {}.".format(list_std3))
print("The standard error of our mean calculation is {}.".format(math_error3))

The mean of the data set is 9.0.
The median of the data set is 10.0.
The variance of the data set is 15.428571428571429.
The standard deviation of the data set is 3.927922024247863.
The standard error of our mean calculation is 1.6035674514745466.


By replacing cousin oliver, whose age was 8, with another child that 1 yr old, we see that the mean (average), of the data set decreases significantly from 10 to 9.  The median stays the same.  The variance increases significantly from 5.43 to 15.43 and the standard deviation increases significantly from 2.52 to 4.24.  This suggests that variation among our data points increased.  Because the variation increased, resulting in our mean being biased towards the younger age numbers, I would not use mean as my preferred method of central tendency in this example.  I would rather go with the median, because its a better representaion of the different age groups.  I would still use standard deviation as my variance method becuase most, if not all, datapoints fall within the standard deviation.

I would not consider the respondants from the sciphi magazine, because the target audience from that magazine don't really have anything to do with the brady bunch.  So I would just take the average of the reponses of the other three magazines: (.20 +.23 + .17)/3 = .20; OR 20%